src/background/walletManager.ts

```diff
@@ -53,30 +53,31 @@ export class WalletManager {
     this.app = appManager
 
     this.existDirectory = fs.existsSync(SHEIKAH_PATH)
+  }
 
+  //  Start running the wallet release and download it when is necessary
+  public async run() {
     if (this.existDirectory) {
+      // Check if latest version is compatible or needs to be downloaded
       try {
         const versionName = fs.readFileSync(
           path.join(SHEIKAH_PATH, VERSION_FILE_NAME),
           'utf8',
         )
         const installedVersion = getVersionFromName(versionName)
+        const latestWitnetRustVersion = await getLatestWitnetRustRelease()
```

Comment:
What will happen if we receive a timeout from Github? We should handle that error and don't update the wallet in this case

Reply:
> What will happen if we receive a timeout from Github? We should handle that error and don't update the wallet in this case

This function is inside a try-catch